WIP: Re-imagining the CCADB Policy (#1)
ryancdickson/www.ccadb.org: policy.md

# CCADB Policy

*Version 2.0, Effective: June 1, 2025*

## Introduction

Several Public Root Store Operators have collaborated to create the Common Certification Authority Database (CCADB), a data repository of certificate and Certification Authority (CA) information. 

CA Owners who wish to be included in a CCADB-participating Root Store ("Root Store") need to disclose and maintain certain information in the CCADB. This policy describes common Root Store Operator requirements related to CA Owner use of the CCADB. This policy does not cover how to obtain write access to the CCADB or how to use the CCADB’s web interface. Additional information covering these topics is available [here](https://www.ccadb.org/cas/).

> Root Store Operators MAY have additional CCADB-related requirements defined in their own Root Store policies. These policies MAY strengthen the requirements described in this policy. Additional information on the policies belonging to the individual Root Store Operators participating in the CCADB can be found [here](https://www.ccadb.org/resources). Questions related to an individual Root Store Operator Policy SHOULD be directed to the operator of that policy using the contact address defined in its policy.

Reviewer: Edit last sentence to read "be directed to the operator of that policy using the contact address designated for that policy."

Author: Addressed in https://github.com/ryancdickson/www.ccadb.org/commit/37a24211fd3cc3909470260c238fcd73c4a40428.

All CCADB disclosures MUST be made freely available and without additional requirements, including, but not limited to, registration, legal agreements, or restrictions on redistribution of the certificates in whole or in part.

All data in the CCADB may be made public or semi-public in a variety of forms. CA Owners SHOULD NOT place any information in the CCADB which they wish to keep confidential. 

Root Store Operators are not obliged to publish any CCADB information.

All client devices that are used to download Personally Identifiable Information from the CCADB SHOULD employ disk-based encryption.

## 2. Contact Information Disclosures

CA Owners MUST provide the following information for a Primary Point of Contact (POC), and at least one other POC, who MAY be a Primary POC:

* Name
* CA Owner-affiliated email address

> CA Owners MAY optionally designate further POCs, supplying an email address for each. CA Owners MUST also supply at least one non-personal contact email aliases which are more likely to continue working as personnel change; these are maintained as part of the CA’s organizational entry.

Reviewer: Should be singular "one non-personal contact email alias that is more likely"

Author: Addressed in https://github.com/ryancdickson/www.ccadb.org/commit/2359c50310e6d8009e5d252b4277eaff4f7ae782.

All Primary POCs SHOULD be authorized to speak for and enter into binding commitments on behalf of the CA(s) that they represent. The Primary POC will be issued a [CA Community license](https://www.ccadb.org/cas/request-access#application-for-ccadb-access) and MAY request additional licenses for other Primary POCs. Primary POCs are responsible for keeping CCADB data accurate for their CA(s).

> Notification of security and audit-related issues MAY be emailed to all Primary POCs, non-personal contact email alias(es), or a combination of these addresses. CA Owners are advised to make sure those addresses reach sufficient people such that they can respond to an issue in an appropriate timeframe.

Reviewer: Strike "Primary" so that it includes all POCs.

Author: Addressed in https://github.com/ryancdickson/www.ccadb.org/commit/c988019d3ca961fcbf31f8e86956a52478ec109b.

> CA Owners MUST maintain up to date contact details in the CCADB. If POC information needs to be updated, the CA Owner MUST submit an [Add/Update Contacts Case](https://www.ccadb.org/cas/contacts).

Reviewer: Maybe add a sentence "A CA Owner's non-personal contact email alias(es) can be updated with an Add/Update Root Request."

Author: Addressed in https://github.com/ryancdickson/www.ccadb.org/commit/c988019d3ca961fcbf31f8e86956a52478ec109b. 

I slightly changed this to:

"A CA Owner's non-personal contact email alias(es) MUST be updated with an [Add/Update Root Request Case](https://www.ccadb.org/cas/updates)." - as I it seems we should prioritize CA Owners changing their own records using the well-defined process rather than the SC changing records on their behalf.

## 3. Certificate Disclosures

> Some Root Store Operators require certificate disclosures to the CCADB.

Reviewer: Delete this sentence?

Author: Addressed in https://github.com/ryancdickson/www.ccadb.org/commit/ab60533a5ad2b3f6529b8f9df7f9cdff43a60213.

CA Owners MUST maintain correct and current information about their root CA certificates. If root CA certificate information needs to be added or updated, the CA Owner MUST submit an [Add/Update Root Request Case](https://www.ccadb.org/cas/updates).

### 3.2 Subordinate CA Certificates

A subordinate CA certificate is a certificate signed by a root CA or another subordinate CA. The CCADB considers the terms "intermediate" and "subordinate" synonymous. 

CA Owners MUST disclose:
- all subordinate CA certificates capable of validating to a certificate included in a Root Store or associated with a CCADB Root Inclusion Request. Disclosure MUST take place within 7 calendar days of issuance and before the subject CA represented in the certificate begins issuing publicly-trusted certificates.
- revocation of all subordinate CA certificates capable of validating to a certificate included in a Root Store or associated with a CCADB Root Inclusion Request within 7 calendar days of revocation.

CA certificates created by cross-signing are considered subordinate CA certificates by Root Store Operators and MUST be disclosed to both the issuer and subject CA Owner PKI hierarchies in the CCADB. 

> For any revoked subordinate CA certificate, each corresponding revocation entry published to a CRL MUST include a reasonCode extension. If the revocation is due to a security concern, the CA Owner MUST file a secure (i.e., "confidential") [Incident Report](https://www.ccadb.org/cas/incident-report) in Bugzilla, with the expectation that a public Incident Report will eventually being filed.

Reviewer: Replace "being" with "be".

Reviewer 2: ```suggestion
For any revoked subordinate CA certificate, each corresponding revocation entry published to a CRL MUST include a reasonCode extension. If the revocation is due to a security concern, the CA Owner MUST file a secure (i.e., "confidential") [Incident Report](https://www.ccadb.org/cas/incident-report) in Bugzilla, with the expectation that a public Incident Report will eventually be filed.
```

## 4. Policy Disclosures

CCADB enables associating PKI policy documents in the records for both root CA certificates and subordinate CA certificates.

CA Owners with either (1) a certificate included in a Root Store or (2) a CA certificate that validates to a certificate included in a Root Store MUST minimally:

- accurately describe the policies and practices of their CA(s) within a Certificate Policy (CP) and corresponding Certification Practice Statement (CPS), or preferably, a single document combined as a CP/CPS.
- ensure the CA policy documents are:
    - freely publicly available for examination.
    - available in an authoritative English language version.

>     - sufficiently detailed to assess the operations of the CA(s) and the compliance with these expectations and those of the Baseline Requirements, and MUST NOT conflict with either of these requirements.

Reviewer: "either of these requirements" is vague. It should say something like, "Sufficiently detailed to assess the operations of the CA(s) and their compliance with the expectations set forth in this Policy, the applicable CA/Browser Forum Baseline Requirements, and any applicable Root Store Operator policies, and MUST NOT conflict with any of the requirements specified therein."

Reviewer 2: ```suggestion
    - sufficiently detailed to assess the operations of the CA(s) and their compliance with the expectations set forth in this Policy, the applicable CA/Browser Forum Baseline Requirements, and any applicable Root Store Operator policies, and MUST NOT conflict with any of the requirements specified therein.
```

To promote simplicity and clarity, all CA policy documents SHOULD be:
- focused on one specific PKI use case (e.g., TLS server authentication, TLS client authentication, S/MIME, Code Signing, etc.), rather than combining multiple use cases into a single document or set of documents.
- comprehensive and consolidated, whenever possible, such that there are not multiple sets of similar yet slightly different policy and practice statements supporting the same PKI use case.
- available in Markdown or AsciiDoc.

> CA Owners MUST strictly adhere to their policy document(s) as disclosed within the CCADB (and not marked as “Superseded”). This extends to all policy documents the CA Owner publishes in relation to its CAs included in a Root Store, such as TSPS documents.

Reviewer: ", including but not limited to Trust Service Provider practice statements and PKI disclosure statements."  ?

Reviewer 2: ```suggestion
CA Owners MUST strictly adhere to their policy document(s) as disclosed within the CCADB (and not marked as “Superseded”). This extends to all policy documents the CA Owner publishes in relation to its CAs included in a Root Store, including but not limited to Trust Service Practice Statement (TSPS) documents and PKI disclosure statements.
```

Before corresponding policy changes are put into practice, CA Owners:
- MUST minimally ensure the updated versions of a CA's policy document(s) are uploaded to their own publicly accessible repository, and
- SHOULD ensure the updated versions of a CA's policy document(s) are submitted to the CCADB within 7 calendar days of the policy document's effective date, but MUST do so within 14 calendar days.

CA Owners MUST:
- submit an [Add/Update Root Request Case](https://www.ccadb.org/cas/updates) to add or update policy documents for root CA certificates stored in the CCADB.

> - add or update  policy documents for subordinate CA certificates directly on the record in CCADB (unless the exceptions stated above apply).

Reviewer: Shouldn't this say something like, "in accordance with the instructions below" ?   What exceptions above apply?

Reviewer 2: ```suggestion
- add or update policy documents for subordinate CA certificates directly on the record in CCADB in accordance with the instructions below (unless the exceptions stated below apply).
```

| Secure Email           | Any certificate issued by the CA (directly or transitively) contains an EKU value of: <br><br>(1) either 1.3.6.1.5.5.7.3.4 (id-kp-emailProtection) or 1.3.6.1.5.5.7.3.0 (anyExtendedKeyUsage), or <br><br> (2) No EKU   | (1) Standard Audit <br><br> (2) NetSec Audit <br><br> (3) S/MIME BR Audit  |

Some Root Store Operators require additional audit disclosures to the CCADB.

Exceptions to providing these audit statement URLs:

* The SHA-256 fingerprint of the certificate is specifically listed as in scope in the audit statements of the parent certificate, and the "Audits Same as Parent" checkbox is checked; or
* The certificate has expired; or
* The certificate has been revoked, and the corresponding record in the CCADB has been updated with the correct revocation status.

CA Owners MUST:
- submit an [Add/Update Root Request Case](https://www.ccadb.org/cas/updates) to add or update audit statements for root CA certificates stored in the CCADB.
- add or update audit statements for subordinate CA certificates directly on the record in the CCADB (unless the exceptions stated above apply).

> The presence of qualifications in an audit statement is not, by itself, generally considered a reason to remove a CA Owner from a Root Store. The purpose of audits is to honestly and thoroughly assess a CA Owner’s compliance with requirements which are necessary to assure a secure and stable ecosystem. Audit findings, including qualifications, can help to identify opportunities for improvement, whether for individual CAR Owners or for the wider industry as a whole.

Reviewer: Minor typo - "CAR owners".

Reviewer 2: ```suggestion
The presence of qualifications in an audit statement is not, by itself, generally considered a reason to remove a CA Owner from a Root Store. The purpose of audits is to honestly and thoroughly assess a CA Owner’s compliance with requirements which are necessary to assure a secure and stable ecosystem. Audit findings, including qualifications, can help to identify opportunities for improvement, whether for individual CA Owners or for the wider industry as a whole.
```

### 5.1 Scope of Certificates Covered by an Audit

CA certificates MUST appear in the Issuer’s annual audit statement, beginning with the statement corresponding to the audit period when the certificate was issued. Those certificates MUST continue to appear in audit statements until having appeared on at least one audit statement following revocation or key destruction (observed and reported by a Qualified Auditor).

> CA certificates created by cross-signing are also considered subordinate CA certificates by Root Store Operators and MUST be included in all relevant audit statements of the entity that possesses the private key that performed the cross-signing.

Reviewer: Do we need to add something like "and unused self-signed root certificates for the same key pair are considered subordinate CA certificates ..." ?

Reviewer 2: I think the fact that we use "CA certificates" instead of root or subordinate has us covered, but what would our definition of an "unused" self-signed root certificate be?

Cross-certificates issued before June 1, 2025 that DO NOT contain an EKU MUST appear in all of the Issuer’s annual audit statements for its corresponding Trust Purposes ([described above](policy#5-policy-disclosures)).

### 5.2 Audit Statement Content

> An authoritative English language version of publicly available audit information MUST be uploaded to the CCADB no later than 92 calendar days of the point-in-time date or the end date of the period of time. In the event of a delay greater than 92 calendar days, the CA Owner MUST provide an explanatory letter signed by the Qualified Auditor. The CCADB warns Root Store Operators when the audit periods are not consecutive.

Reviewer: Maybe find another place for this sentence, "The CCADB warns Root Store Operators when the audit periods are not consecutive."  Or delete it and start the paragraph out with the following:  "Audit periods MUST be consecutive, with no gaps."

Reviewer 2: ```suggestion
Audit period MUST be consecutive, with no gaps in time between audits. An authoritative English language version of publicly available audit information MUST be uploaded to the CCADB no later than 92 calendar days of the point-in-time date or the end date of the period of time. In the event of a delay greater than 92 calendar days, the CA Owner MUST provide an explanatory letter signed by the Qualified Auditor. 
```

Reports uploaded to the CCADB MUST be publicly available and contain at least the following clearly-labeled text-searchable information: 

1. Full name of the CA Owner or corresponding Affiliate (e.g., organization providing external RA services) that was audited;
2. Name and address of the organization performing the audit;
3. Qualifications of the team performing the audit;

> 4. The physical locations that were or were not audited;

Reviewer: What about "Physical locations for CA operations that were ..." ?

Reviewer 2: ```suggestion
4. Physical locations for CA operations that were or were not audited;
```

5. SHA-256 fingerprint of each root and subordinate CA certificate that was in scope of the audit (see format specifications below);
6. Full names and version numbers of the audit standards that were used during the audit;
7. List of the CA Owner's applicable policy documents (with version numbers and publication dates) referenced during the audit;
8. Whether the audit is for a period of time or a point in time;
9. Start date and end date of the period that was audited, for those that cover a period of time (this is not the period the auditor was on-site);
10. Point-in-time date, for those that are for a point in time;

> 11. Date the audit statement was written, which will necessarily be after the audit period end date or point-in-time date (see date format specifications below);

Reviewer: What about "Date that the audit statement was written or signed" ?  Also, for ETSI they sometimes re-issue a subsequent version of the same audit. In that case, there might be two or three "issuance" dates, and I prefer to allow CAs to populate the CCADB with the initial date of the ETSI AAL so that ALV does not alert that the AAL is late.  I might add something like, "For revised or corrected versions of audit statements made to address errors, omissions, or formatting to a previously completed audit statement, the CA Owner may enter the date of the initial issuance, provided that the scope and audit period covered by the revised audit statement remain unchanged. In such cases, the audit statement should contain the initial issuance date and the subsequent issuance date and the CA Owner should provide the URL of the most recent version and include a comment in the case explaining the revision date and the reason for reissuance of the audit statement."

Reviewer 2: Regarding the first suggestion, I'm inclined to leave it as is since the current languages matches section 8.6 number 9 of the current TLS BRs. A longer term goal should be to harmonize this section and the BRs for simplicity. 

Regarding the second suggestion, we also know that WebTrust will reissue audits to correct errors and **not** change the version, date, or Seal URL, which is less than ideal IMO, but according to the auditing standards they cannot change the date (or something along those lines that I'm not fully remembering right now). We have an open unresolved thread with the WTTF on the topic. I personally do not mind the ALV error presenting itself to communicate unusual audit statement behavior and thus signaling the need for further investigation. I also prefer to not promote errors and omissions. All of that said, if you feel really strongly about the suggested language - I'm okay to add it.

The presence of qualifications in an audit statement is not, by itself, generally considered a reason to remove a CA Owner from a Root Store. The purpose of audits is to honestly and thoroughly assess a CA Owner’s compliance with requirements which are necessary to assure a secure and stable ecosystem. Audit findings, including qualifications, can help to identify opportunities for improvement, whether for individual CAR Owners or for the wider industry as a whole.

### 5.1 Scope of Certificates Covered by an Audit

CA certificates MUST appear in the Issuer’s annual audit statement, beginning with the statement corresponding to the audit period when the certificate was issued. Those certificates MUST continue to appear in audit statements until having appeared on at least one audit statement following revocation or key destruction (observed and reported by a Qualified Auditor).

> 

Reviewer 2: ```suggestion
Self-signed certificates that share a Subject + SPKI with a root certificate that is, or was, included in a Root Store are treated by Root Store Operators as subordinate CA certificates because they chain up to an included root, so these certificates MUST also be listed in the applicable audit statements according to the [Derived Trust Bits](https://www.ccadb.org/cas/fields#formula-fields) field.
```
@BenWilson-Mozilla maybe adding this statement from the https://www.ccadb.org/cas/alv page better addresses your comment below.

- 2.23.140.1.5.4.2
- 2.23.140.1.5.4.3

#### 5.2.1 ETSI

> Audits conducted by an accredited Conformity Assessment Body (CAB) MUST have their Audit Attestation Letter (AAL) uploaded to the CAB’s website. CA Owners provide the URL to the AAL on the CAB’s website, and ALV will verify those URLs against a list of approved CAB websites. ETSI AALs MUST follow the latest version of the AAL template on the ACAB'c [website](https://www.acab-c.com/downloads/).

Reviewer: Alternative language:  "An Audit Attestation Letter (AAL) issued by an accredited Conformity Assessment Body (CAB) MUST be downloadable from the CAB’s designated repository for such AALs, which must be on the CCADB's list of approved websites. CA Owners must populate the CCADB with the URL where such AAL can be downloaded over https."  Then we could delete "ALV will verify those URLs against a list of approved CAB websites." ?

### 6.2 Certificate Revocation List Disclosures

For each unexpired and unrevoked CA certificate disclosed to the CCADB and within 7 days of the corresponding CA issuing its first certificate, CA Owners MUST disclose either:
- the URL of a full and complete Certificate Revocation List (CRL); or
- a JSON Array of Partitioned CRL URLs.

URLs:
- MUST match exactly as they appear in the certificates issued by the corresponding CA.

If populating a full and complete CRL URL: 
- the corresponding CRL SHOULD NOT contain an 'Issuing Distribution Point' extension.
- the JSON Array of Partitioned CRL URLs field MUST be empty.

If populating a JSON Array of Partitioned CRL URLs: 
- CA Owners MUST ensure that each corresponding CRL contains a critical 'Issuing Distribution Point' extension and the 'distributionPoint' field of the extension MUST include a 'UniformResourceIdentifier'. The value of the UniformResourceIdentifier MUST exactly match a URL, from which the CRL was accessed, present in the CCADB record associated with the CA certificate.
- the full and complete CRL URL MUST be empty.

> Under normal operating conditions, the CRL URLs provided by CAs in this section MUST be available such that relying parties are able to successfully retrieve the current CRL every 4 hours.

Reviewer: "in **_accordance with_** this section"

Author: Addressed in https://github.com/ryancdickson/www.ccadb.org/commit/52c7b33fdd3fe256ff3cde154b48fb18f37df843.

A single self-assessment MAY cover multiple CAs operating under both the same CP and CPS(s), or combined CP/CPS. CAs not operated under the same CP and CPS(s) or combined CP/CPS MUST be covered in a separate self-assessment.

The self-assessment submission date is determined by the earliest appearing "BR Audit Period End Date" field specified in any of the CA Owner’s "CA Owner/Certificate" CCADB root records that are included in one or more Root Stores. 

An annual self-assessment MUST be completed and submitted to the CCADB within 92 calendar days from the CA Owner's earliest appearing root record "BR Audit Period End Date." CA Owners SHOULD submit the self-assessment to the CCADB at the same time as uploading audit reports.

## 7. Mailshots

> From time to time, a Root Store Operator MAY use the CCADB to send information to CA Owners. Mailshots will be sent to at least the Primary POC(s) and the email aliases, and may also go to one or more of the other POCs; the exact recipient list is defined by the Root Store Operator sending the information.

Reviewer: Revise "and the email aliases" to "and the CA Owners' email aliases".

Author: Addressed in https://github.com/ryancdickson/www.ccadb.org/commit/2ff22b3e418c7d484b802497b67cc06bea7c8202.

Mozilla expects the timely remediation of the problems that caused or gave rise to an incident. In response to incidents, Mozilla MAY further require that the CA operator submit a plan of action with milestones or submit one or more additional audits to provide sufficient assurance that the incident has been remediated. Such audits MAY be expected sooner than the CA operator’s next scheduled audit, and thus MAY be expected to be for a period less than a year.

--- 

[X] Audit reports that are being supplied to maintain a certificate within the Mozilla root store MUST be provided to Mozilla via the CCADB within three months of the point-in-time date or the end date of the period of time.

----

[X] The publicly-available documentation relating to each audit MUST contain the information required by section 5.1 of the CCADB Policy (v.1.3.1) and the CA locations that were or were not audited. Audit reports MUST also contain or be accompanied by the name of the lead auditor and qualifications of the team performing the audit, as required by section 3.2.

> If Mozilla determines that an audit provided does not meet the requirements of this policy, then Mozilla MAY require that the CA operator obtain a new audit, at the CA operator's expense, for the period of time in question. Additionally, depending on the nature of concerns with the audit, Mozilla MAY require that the CA operator obtain such an audit from a new auditor.

Reviewer: Does the CCADB policy revision state that a CA owner might have to obtain a new audit?

Author: Addressed in https://github.com/ryancdickson/www.ccadb.org/commit/e3ea8302985642c1294b03289c0bf8d594cb5d2e.

[x] All disclosure MUST be made freely available and without additional requirements, including, but not limited to, registration, legal agreements, or restrictions on redistribution of the certificates in whole or in part.

> [ ] We recognize that technically constraining intermediate certificates as described above may not be practical in some cases. All certificates that are capable of being used to issue new certificates, that are not technically constrained, and that directly or transitively chain to a certificate included in Mozilla’s root store MUST be audited in accordance with this policy, and the corresponding audit statements MUST be disclosed in the CCADB according to section 5 of the CCADB Policy. If the CA operator has a currently valid audit report at the time of creation of the intermediate certificate, then the new intermediate certificate MUST appear on the CA operator's next periodic audit reports.

Reviewer: Do we need to add something similar to this re: "If the CA operator has a currently valid audit report at the time of creation of the intermediate certificate, then the new intermediate certificate MUST appear on the CA operator's next periodic audit reports." ?

Author: What if we included this as an exception? One possible framing highlighted with the bold text below.

Exceptions to providing these audit statement URLs:

- The SHA-256 fingerprint of the certificate is specifically listed as in scope in the audit statements of the parent certificate, and the "Audits Same as Parent" checkbox is checked; or
- The certificate has expired; or
- The certificate has been revoked, and the corresponding record in the CCADB has been updated with the correct revocation status; or
- **The certificate was issued after the audit period of the most recently performed audit. In such cases, the certificate MUST be included in the scope of the CA Owner's next periodic audit.**

Comments and suggestions welcome.

---

[ ] Additionally, CA operators applying for inclusion of new root certificates with the websites trust bit enabled MUST demonstrate support for at least one automated method of certificate issuance for each type of TLS certificate (EV, OV, DV, IV) intended to be issued under the root certificate being requested for inclusion. This means (1) automated domain control validation, as defined in the TLS BRs; and (2) automated certificate issuance and retrieval processes. Such automated methods MUST minimize hands-on human input during routine certificate issuance and renewal processes and comply with the TLS BRs, and EV Guidelines, if applicable. Acceptable "hands-on" input includes initial software setup, configuration, updates, and identity verification where required. CA operators MUST renew test certificates using such capability at least every 30 days to demonstrate compliance with these automation requirements. The test certificates MUST be served by publicly accessible websites, and the URL for each test site MUST be disclosed in the CCADB.

--- 

> [X] A failure to provide notifications or updates in the CCADB or as otherwise required in a timely manner SHALL also be grounds for disabling a CA operator’s root certificates or removing them from Mozilla's root store. For this policy and the CCADB policies, "a timely manner" means within 30 days of when the appropriate data or documentation becomes available to the CA operator, unless a Mozilla policy document specifies a different rule.

Reviewer: A Github issue might need to be filed in the Mozilla repository to remove this 30-day language from the MRSP.

Author: ACK. We'll leave that for you to manage! The same might apply for other elements of the MRSP that might now be redundant with the CCADB policy.
